src/apify.js

```diff
@@ -0,0 +1,961 @@
+     *
+     * ```javascript
+     *
+     * // Returns initialized proxy configuration class
+     * const proxyConfiguration = await Apify.createProxyConfiguration({
+     *     groups: ['GROUP1', 'GROUP2'] // List of Apify proxy groups
+     *     countryCode: 'US'
+     * });
+     *
+     * const crawler = new Apify.CheerioCrawler({
+     *   // ...
+     *   proxyConfiguration,
+     *   handlePageFunction: ({ proxyInfo }) => {
+     *       const usedProxyUrl = proxyInfo.url; // Getting the proxy URL
+     *   }
+     * })
+     *
+     * ```
+     *
+     * For compatibility with existing Actor Input UI (Input Schema), this function
+     * returns `undefined` when the following object is passed as `proxyConfigurationOptions`.
+     *
+     * ```
+     * { useApifyProxy: false }
+     * ```
+     *
+     * @param {ProxyConfigurationOptions} [proxyConfigurationOptions]
+     * @returns {Promise<ProxyConfiguration|undefined>}
+     */
+    async createProxyConfiguration(proxyConfigurationOptions = {}) {
+        // Compatibility fix for Input UI where proxy: None returns { useApifyProxy: false }
+        // Without this, it would cause proxy to use the zero config / auto mode.
+        const dontUseApifyProxy = proxyConfigurationOptions.useApifyProxy === false;
+        const dontUseCustomProxies = !proxyConfigurationOptions.proxyUrls;
+
+        if (dontUseApifyProxy && dontUseCustomProxies) {
+            return;
+        }
+
+        const proxyConfiguration = new ProxyConfiguration(proxyConfigurationOptions, this.config);
+        await proxyConfiguration.initialize();
+
+        return proxyConfiguration;
+    }
+
+    /**
+     * Returns a new {@link ApifyEnv} object which contains information parsed from all the `APIFY_XXX` environment variables.
+     *
+     * For the list of the `APIFY_XXX` environment variables, see
+     * [Actor documentation](https://docs.apify.com/actor/run#environment-variables).
+     * If some of the variables are not defined or are invalid, the corresponding value in the resulting object will be null.
+     * @returns {ApifyEnv}
+     */
+    getEnv() {
+        return getEnv();
+    }
+
+    /**
+     * Returns a new instance of the Apify API client. The `ApifyClient` class is provided
+     * by the <a href="https://www.npmjs.com/package/apify-client" target="_blank">apify-client</a>
+     * NPM package, and it is automatically configured using the `APIFY_API_BASE_URL`, and `APIFY_TOKEN`
+     * environment variables. You can override the token via the available options. That's useful
+     * if you want to use the client as a different Apify user than the SDK internals are using.
+     *
+     * @param {object} [options]
+     * @param {string} [options.token]
+     * @param {string} [options.maxRetries]
+     * @param {string} [options.minDelayBetweenRetriesMillis]
+     * @return {ApifyClient}
+     */
+    newClient(options = {}) {
+        return this.config.createClient(options);
```

Comment:
The way it was written before was relying on the global state in utils (which was impossible to configure without env vars), this is in my eyes much better, at least it clearly shows the dependencies, not like before. 

If we want to have singletons, we should have a DI container to managed this, instead of relying on require time instances. The `Configuration` object can serve as one too - if we really need singletons now, will rewrite it that way. The end goal should be to get rid of the static helper methods we have now. 

We could also use `AsyncLocalStorage` for this, if we ditch node 10 (which will reach EOL in few weeks anyway). That way we could run have a shared storage for the context of `main` method.